app-audiotest/main.c

```diff
@@ -0,0 +1,68 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "mach_defines.h"
+#include "gfx_load.h"
+#include "cache.h"
+
+//hardware addresses I really need to dump into an include or something...
+extern volatile uint32_t MISC[];
+#define MISC_REG(i) MISC[(i)/4]
+
+extern volatile uint32_t SYNTH[];
+#define SYNTHREG(i) SYNTH[i/4]
+
+static inline void button_wait(){
+	//Wait until all buttons are released
+	while (MISC_REG(MISC_BTN_REG)) ;
+	//Wait until button A is pressed
+	while ((MISC_REG(MISC_BTN_REG) & BUTTON_A)==0) ;
+}
+
+static inline void pause(void){
+	for (volatile uint32_t i=0; i<0x0003fff0; i++){;}
+}
+
+void main(int argc, char **argv) {
+	//We're running in app context. We have full control over the badge and can do with the hardware what we want. As
+	//soon as main() returns, however, we will go back to the IPL.
+
```

Comment:
Can you document this, e.g. using symbolic names for the regs and values? This looks like you reverse engineered it from some alien technology...

Reply:
Yes, I need to write both C code in there, and register macros.  
It's all in text ATM in audio_registers.html.  Decyphers the alien technology.  :)  But yeah, macros are on the todo list.